app/models/course.rb

```diff
@@ -171,10 +173,32 @@ def initialize_duplicate(duplicator, other)
     self.levels = duplicator.duplicate(other.levels)
     self.achievements = duplicator.duplicate(other.achievements)
     self.forums = duplicator.duplicate(other.forums)
+
+    # This updates ids in the course settings after duplicated objects have been saved
+    # and the ids are available.
+    self.define_singleton_method(:update_settings_after_duplication) do
```

Comment:
Should this be put in the duplication service?  that is, handle the settings and other cleanups after the course is created in the duplication service.